hivemind_core/protocol.py

```
@@ -224,15 +212,25 @@ class HiveMindListenerProtocol:
+                                                                  agent_protocol=self.agent_protocol)
+        else:
+            self.binary_data_protocol.hm_protocol = self
 
     def get_bus(self, client: HiveMindClientConnection) -> Union[FakeBus, MessageBusClient]:
         # allow subclasses to use dedicated bus per client
         return self.agent_protocol.bus
 
     def handle_new_client(self, client: HiveMindClientConnection):
+        try:
+            self.callbacks.on_connect(client)
+        except:
```

Ruff (0.8.2)

Do not use bare `except`

(E722)